src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -263,6 +265,12 @@ ShouldUseSubqueryPushDown(Query *originalQuery, Query *rewrittenQuery)
 		return true;
 	}
 
+	whereClauseList = WhereClauseList(rewrittenQuery->jointree);
```

Comment:
Needs a comment/justification why we're allowing this through the pushdown planner. Something like in the commit message.

Reply:
added

Reply:
`recursive planner` should be `pushdown planner` I think.